packages/engine/src/scene/components/TextComponent.tsx

```diff
@@ -0,0 +1,411 @@
+      text.troikaMesh.value.letterSpacing = text.letterSpacing.value
+      text.troikaMesh.value.lineHeight = text.lineHeight.value
+      text.troikaMesh.value.direction = text.textDirection.value
+      // Update the font properties
+      text.troikaMesh.value.font = text.font.value
+      text.troikaMesh.value.fontSize = text.fontSize.value
+      text.troikaMesh.value.color = text.fontColor.value.getHex()
+      switch (text.fontMaterial.value) {
+        case FontMaterialKind.Basic:
+          text.troikaMesh.value.material = new MeshBasicMaterial()
+          break
+        case FontMaterialKind.Standard:
+          text.troikaMesh.value.material = new MeshStandardMaterial()
+          break
+      }
+      text.troikaMesh.value.outlineOpacity = text.outlineOpacity.value / 100
+      text.troikaMesh.value.outlineWidth = `${text.outlineWidth.value}%`
+      text.troikaMesh.value.outlineBlur = `${text.outlineBlur.value}%`
+      text.troikaMesh.value.outlineOffsetX = `${text.outlineOffset.x.value}%`
+      text.troikaMesh.value.outlineOffsetY = `${text.outlineOffset.y.value}%`
+      text.troikaMesh.value.outlineColor = text.outlineColor.value.getHex()
+      text.troikaMesh.value.strokeOpacity = text.strokeOpacity.value / 100
+      text.troikaMesh.value.strokeWidth = `${text.strokeWidth.value}%`
+      text.troikaMesh.value.strokeColor = text.strokeColor.value.getHex()
+      // Update the Advanced configuration propertiess
+      text.troikaMesh.value.orientation = text.textOrientation.value
+      ;(text.troikaMesh.value.clipRect = text.clipActive.value
+        ? [
+            // Send as [minX, minY, maxX, maxY] :Array<number>
+            text.clipRectMin.x.value,
+            text.clipRectMin.y.value,
+            text.clipRectMax.x.value,
+            text.clipRectMax.x.value
+          ]
+        : []),
+        (text.troikaMesh.value.gpuAccelerateSDF = text.gpuAccelerated.value)
+      text.troikaMesh.value.sdfGlyphSize = Math.pow(2, text.glyphResolution.value)
+      text.troikaMesh.value.glyphGeometryDetail = text.glyphDetail.value
+      // Order troika to synchronize the mesh
+      text.troikaMesh.value.sync()
+    }, [
+      text.text,
+      text.textOpacity,
+      text.textIndent,
+      text.textAlign,
+      text.textWrap,
+      text.textWrapKind,
+      text.textAnchor,
+      text.textCurveRadius,
+      text.textDepthOffset,
+      text.textWidth,
+      text.lineHeight,
+      text.letterSpacing,
+      text.textDirection,
+      text.fontSize,
+      text.fontColor,
+      text.fontMaterial,
+      text.outlineOpacity,
+      text.outlineWidth,
+      text.outlineBlur,
+      text.outlineOffset,
+      text.outlineColor,
+      text.strokeOpacity,
+      text.strokeWidth,
+      text.strokeColor,
+      text.textOrientation,
+      text.clipActive,
+      text.clipRectMin,
+      text.clipRectMax,
+      text.gpuAccelerated,
+      text.glyphResolution,
+      text.glyphDetail
+    ])
+
+    return null
```

Comment:
Instead of returning null. Return a function that runs 
`text.troikaMesh.dispose()`

Reply:
I think its resolved with https://github.com/EtherealEngine/etherealengine/pull/9341/commits/f2b142afc1c0aa99d08aaa31eb3685b511157821
Let me know if something is missing.